$OpenBSD$
Add _SAMBA4 sufffix to generated routines to de-conflict with base system
Heimdal installation.
--- source4/heimdal/lib/asn1/gen_decode.c.orig	Sat May 18 20:11:12 2013
+++ source4/heimdal/lib/asn1/gen_decode.c	Sat May 18 20:24:15 2013
@@ -41,14 +41,14 @@ decode_primitive (const char *typename, const char *na
 {
 #if 0
     fprintf (codefile,
-	     "e = decode_%s(p, len, %s, &l);\n"
+	     "e = decode_%s_SAMBA4(p, len, %s, &l);\n"
 	     "%s;\n",
 	     typename,
 	     name,
 	     forwstr);
 #else
     fprintf (codefile,
-	     "e = der_get_%s(p, len, %s, &l);\n"
+	     "e = der_get_%s_SAMBA4(p, len, %s, &l);\n"
 	     "if(e) %s;\np += l; len -= l; ret += l;\n",
 	     typename,
 	     name,
@@ -220,7 +220,7 @@ decode_type (const char *name, const Type *t, int opti
 		    "if (%s == NULL) %s;\n",
 		    name, name, name, forwstr);
 	fprintf (codefile,
-		 "e = decode_%s(p, len, %s, &l);\n",
+		 "e = decode_%s_SAMBA4(p, len, %s, &l);\n",
 		 t->symbol->gen_name, name);
 	if (optional) {
 	    fprintf (codefile,
@@ -666,7 +666,7 @@ generate_type_decode (const Symbol *s)
     int preserve = preserve_type(s->name) ? TRUE : FALSE;
 
     fprintf (codefile, "int ASN1CALL\n"
-	     "decode_%s(const unsigned char *p HEIMDAL_UNUSED_ATTRIBUTE,"
+	     "decode_%s_SAMBA4(const unsigned char *p HEIMDAL_UNUSED_ATTRIBUTE,"
 	     " size_t len HEIMDAL_UNUSED_ATTRIBUTE, %s *data, size_t *size)\n"
 	     "{\n",
 	     s->gen_name, s->gen_name);
@@ -720,7 +720,7 @@ generate_type_decode (const Symbol *s)
 		 "return 0;\n");
 	fprintf (codefile,
 		 "fail:\n"
-		 "free_%s(data);\n"
+		 "free_%s_SAMBA4(data);\n"
 		 "return e;\n",
 		 s->gen_name);
 	break;
